shell/common/isolate_configuration.h

```diff
@@ -19,30 +19,139 @@
+  ///             where snapshot resolution in extremely expensive, embedders
+  ///             attempt to resolve snapshots on worker thread(s) and return
+  ///             the future of the promise of snapshot resolution to this
+  ///             method. That way, snapshot resolution begins well before
+  ///             isolate launch is attempted by the engine.
+  ///
+  /// @param[in]  kernel_pieces  The list of futures to Dart kernel snapshots.
+  ///
+  /// @return     A JIT isolate configuration.
+  ///
   static std::unique_ptr<IsolateConfiguration> CreateForKernelList(
       std::vector<std::future<std::unique_ptr<const fml::Mapping>>>
           kernel_pieces);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Creates a JIT isolate configuration using the specified
+  ///             snapshot. This is a convenience method for the
+  ///             `CreateForKernelList` method that takes a list of futures to
+  ///             Dart kernel snapshots.
+  ///
+  /// @see        CreateForKernelList()
+  ///
+  /// @param[in]  kernel  The kernel shapshot.
+  ///
+  /// @return     A JIT isolate configuration.
+  ///
+  static std::unique_ptr<IsolateConfiguration> CreateForKernel(
+      std::unique_ptr<const fml::Mapping> kernel);
+
+  //----------------------------------------------------------------------------
+  /// @brief      Create a JIT isolate configuration using the specified
+  ///              snapshots. This is a convenience method for the
+  ///             `CreateForKernelList` method that takes a list of futures to
+  ///             Dart kernel snapshots.
+  ///
+  /// @see        CreateForKernelList()
+  ///
+  /// @param[in]  kernel_pieces  The kernel pieces
+  ///
+  /// @return     { description_of_the_return_value }
+  ///
   static std::unique_ptr<IsolateConfiguration> CreateForKernelList(
       std::vector<std::unique_ptr<const fml::Mapping>> kernel_pieces);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Create an isolate configuration. This has no threading
+  /// restrictions.
+  ///
   IsolateConfiguration();
 
+  //----------------------------------------------------------------------------
+  /// @brief      Destroy an isolate configuration. This has no threading
+  ///             restrictions and may be collection of configurations may occur
+  ///             on any thread (and usually happens on an internal VM managed
+  ///             thread pool thread).
+  ///
   virtual ~IsolateConfiguration();
 
+  //----------------------------------------------------------------------------
+  /// @brief      When an isolate is created and sufficiently initialized to
+  ///             move it into the `DartIsolate::Phase::LibrariesSetup` phase,
+  ///             this method is invoked on the isolate to then move the isolate
+  ///             into the `DartIsolate::Phase::Ready` phase. Then isolate's
```

Comment:
Not sure, but do you think it scans a little better with the clauses of the sentence reversed to put the action up front?
```
Moves the Isolate to the `DartIsolate::Phase::Ready` phase
from the `DartIsolate::Phase::LibrariesSetup` phase. The
isolate's main entrypoint is then invoked to move it into
the...
```

Reply:
I reworded it but then it was not clear who was making the move. Not sure.